soweego/linker/evaluate.py

```diff
@@ -75,58 +134,93 @@ def _compute_performance(test_index, predictions, test_vectors_size):
     return precision, recall, fscore, confusion_matrix
 
 
-def average_k_fold(classifier, catalog, entity, binarize, dir_io, k=5):
+def nested_k_fold_with_grid_search(classifier, param_grid, catalog, entity, k, scoring, dir_io, **kwargs):
+    if classifier is constants.SINGLE_LAYER_PERCEPTRON:
```

Comment:
Probably after merging with master, the `mlp` classifier should be considered as well

```suggestion
    if classifier in (constants.SINGLE_LAYER_PERCEPTRON, 
                      constants.MULTILAYER_CLASSIFIER):
```